hw7/Personality.java

```diff
@@ -0,0 +1,108 @@
+import java.util.*;
+import java.io.*;
+//read name, read original result AB; convert to AB counts two arrays
+//convert to percentage of B; map to personality type;
+//add intro; swap constant for four;
+//check coding style
+
+public class Personality {
+    public static final int arraySize = 4;
+    //main method reads from original data, capturing name and line
+    public static void main(String[] args) throws FileNotFoundException {
+        //print program intro
+        System.out.print("This program processes a file of answers to the\nKeirsey Temperament Sorter.  It converts the \nvarious A and B answers for each person into \na sequence of B-percentages and then into a \nfour-letter personality type.\n");
+        System.out.println();
+        //read from input
+        System.out.print("input file name? ");
+        Scanner console = new Scanner(System.in);
+        String inputFile = console.next();
+        Scanner input = new Scanner (new File (inputFile));
+        //create output
+        System.out.print("output file name? ");
+        String outputFile = console.next();
+        PrintStream output = new PrintStream(new File (outputFile));
+        while (input.hasNextLine()) {
+            String name = input.nextLine();
+            String line = input.nextLine();
+            output.print(name + ": ");
+            int[] countBPercent = bPecent(line);
+            output.print(Arrays.toString(countBPercent));
+            output.print(" = " );
+            String[] type = typeTest (countBPercent);
+            print(output, type);
```

Comment:
-0.25
1. I would suggest change the return type of `typeTest` to String, and return the personality type, instead of passing it to another method for additional processing.
2. `print` on line 32 is a bad name